### ISSUE TITLE
private-intra-doc-links incorrectly shows error for public type alias to private wrapper type

### ISSUE THREAD

**Issue by Manishearth** [rust-lang/rust#132952]
```rust
pub struct WholeNumber(u32);

pub use bar::Integer;
mod foo {
    // This is private
    pub struct Signed<T>(pub T);
}

mod bar {
    /// If you want to do things, try [`Integer::do_thing()`]
    pub type Integer = crate::foo::Signed<crate::WholeNumber>;

    impl Integer {
        pub fn do_thing() {}
    }
}
```

Shows

```
warning: public documentation for `Integer` links to private item `Integer::do_thing`
  --> src/lib.rs:10:41
   |
10 |     /// If you want to do things, try [`Integer::do_thing()`]
   |                                         ^^^^^^^^^^^^^^^^^^^ this item is private
   |
   = note: this link will resolve properly if you pass `--document-private-items`
   = note: `#[warn(rustdoc::private_intra_doc_links)]` on by default
```

under `cargo doc`.

This is (a) incorrect: `Integer::do_thing()` is public and accessible even if `Signed` isn't and (b) misleading: the problem is not the publicness of `do_thing()` but rather the fact that `Signed` is inaccessible. The diagnostic is confusing and doesn't help fix the problem.

I think this should probably be considered a false positive, but either way, the diagnostic ought to be be clearer about what needs to be fixed here.

**Comment by sosthene-nitrokey** [comment]
I encountered the same issue.

Here's a reproducer that also proves that it's publicly visible, and shows another issue: If `do_thing` is defined as part of an `impl` block on the `Signed` type, it does not show up in the docs of `Integer`:

```rust
pub struct WholeNumber;

pub use bar::Integer;
mod foo {
    // This is private
    pub struct Signed<T>(pub T);

    impl<T> Signed<T> {
        pub fn do_thing() {}
    }
}

mod bar {
    /// If you want to do things, try [`Integer::do_thing()`]
    ///
    /// ```
    /// use repro_rustdoc::Integer;
    /// Integer::do_thing();
    /// ```
    pub type Integer = crate::foo::Signed<crate::WholeNumber>;
}
```

In the docs, `do_thing` does not show up (while it does with the first reproducer).


![Image](https://github.com/user-attachments/assets/5bc221d7-288d-4c9d-99c3-cfc546967e77)
